src/modules/launcher/Plugins/Community.PowerToys.Run.Plugin.WebSearch/Main.cs

```diff
@@ -0,0 +1,412 @@
+                    || input.EndsWith("://", StringComparison.CurrentCulture)
+                    || input.All(char.IsDigit))
+                {
+                    return false;
+                }
+
+                try
+                {
+                    _ = new UriBuilder(input);
+                }
+                catch (UriFormatException)
+                {
+                    return false;
+                }
+
+                return true;
+            }
+        }
+
+        private bool IsDefaultBrowserSet()
+        {
+            return !string.IsNullOrEmpty(_browserPath);
+        }
+
+        public void Init(PluginInitContext context)
+        {
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+            _context.API.ThemeChanged += OnThemeChanged;
+            UpdateIconPath(_context.API.GetCurrentTheme());
+            UpdateBrowserIconPath(_context.API.GetCurrentTheme());
+        }
+
+        public string GetTranslatedPluginTitle()
+        {
+            return Properties.Resources.plugin_name;
+        }
+
+        public string GetTranslatedPluginDescription()
+        {
+            return Properties.Resources.plugin_description;
+        }
+
+        private void OnThemeChanged(Theme oldtheme, Theme newTheme)
+        {
+            UpdateIconPath(newTheme);
+            UpdateBrowserIconPath(newTheme);
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage(
+            "Design",
+            "CA1031:Do not catch general exception types",
+            Justification = "We want to keep the process alive but will log the exception")]
+        private void UpdateBrowserIconPath(Theme newTheme)
+        {
+            try
+            {
+                string progId = GetRegistryValue(
+                    "HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\Shell\\Associations\\UrlAssociations\\http\\UserChoice",
+                    "ProgId");
+
+                if (progId.StartsWith("Opera", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Opera user preferences file:
+                    string prefFile;
+
+                    if (progId.Contains("GX", StringComparison.OrdinalIgnoreCase))
+                    {
+                        _browserName = "Opera GX";
+                        prefFile = System.IO.File.ReadAllText($"{Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)}\\Opera Software\\Opera GX Stable\\Preferences");
+                    }
+                    else
+                    {
+                        _browserName = "Opera";
+                        prefFile = System.IO.File.ReadAllText($"{Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)}\\Opera Software\\Opera Stable\\Preferences");
+                    }
+
+                    string url = ExtraStringMethods.GetJSONStringValue(prefFile, new string[] { "default_search_provider_data", "template_url_data", "url" });
```

Comment:
This feels breaky.  Why isn't this in a class

Reply:
what do you mean breaky? which part?

Reply:
manually parsing json versus using a json parser.

https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-use-dom-utf8jsonreader-utf8jsonwriter?pivots=dotnet-6-0#use-jsondocument

This is what i would have done chances are.

Reply:
welp my googling skills failed me

Reply:
this is why we have code reviews and conversations.  I'm always learning stuff.  I learned tons of stuff recently redoing our pipeline build agent